lib/jekyll-redirect-from/redirector.rb

```diff
@@ -71,7 +71,14 @@ def redirect_url(site, item)
     end
 
     def redirect_prefix(site)
-      config_github_url(site) || config_url(site)
+      config_url_and_baseurl(site) || config_github_url(site) || config_baseurl(site) || ""
+    end
+
+    def config_url_and_baseurl(site)
+      return unless config_url(site)
+
+      prefix = File.join(config_url(site).to_s, config_baseurl(site).to_s)
```

Comment:
`URI.join` + `to_s` if so, but I don't think we want to use `URI.join`, because it rules out relative URLs (see earlier discussion at https://github.com/jekyll/jekyll-redirect-from/pull/109/files#r67073653). One could think that it wouldn't be an issue as we return early if `site.url` is `nil`, but if someone would configure their `site.url` to be empty (`""`), we would blow up with `URI::BadURIError: both URI are relative`. I think it's better if we no-op then. I will push specs covering that.